src/test/scala/is/hail/expr/ir/PruneSuite.scala

```diff
@@ -435,11 +435,17 @@ class PruneSuite extends SparkSuite {
   }
 
   @Test def testTableAggregateMemo() {
-    checkMemo(TableAggregate(tab, GetField(Ref("global", tab.typ.globalType), "g1")),
-      TInt32(),
+    checkMemo(TableAggregate(tab, tableRefBoolean(tab.typ, "global.g1")),
+      TBoolean(),
       Array(subsetTable(tab.typ, "global.g1"), null))
   }
 
+  @Test def testMatrixAggregateMemo() {
+    checkMemo(MatrixAggregate(mat, matrixRefBoolean(mat.typ, "global.g1")),
+      TBoolean(),
+      Array(subsetMatrixTable(mat.typ, "global.g1"), null))
```

Comment:
the null here is saying that we don't care what the type of the field reference of `global.g1` is?

Reply:
This is saying the only field we care about is global.g1. checkMemo expects an array of the same length as ir.children, and the null mean s we don't care about the query IR, only the child IR.

Reply:
👌